lib/qam.pm

```diff
@@ -140,15 +140,26 @@ sub advance_installer_window {
     my $build = get_var('BUILD');
 
     send_key $cmd{next};
-    die 'Unable to create repository' if check_screen('unable-to-create-repo', 5);
+    my %handlers;
+
+    $handlers{$screenName} = sub { 1 };
+    $handlers{'unable-to-create-repo'} = sub {
+        die 'Unable to create repository';
+    };
+    $handlers{'cannot-access-installation-media'} = sub {
+        send_key "alt-y";
+        return 0;
+    };
+
     if ($build =~ m/^MR:/) {
-        if (check_screen("import-untrusted-gpg-key", 20)) {
+        $handlers{'import-untrusted-gpg-key'} = sub {
             send_key "alt-t";
-        }
+            return 0;
+        };
     }
-    unless (check_screen "$screenName", 60) {
-        my $key = check_screen('cannot-access-installation-media') ? "alt-y" : "$cmd{next}";
-        send_key_until_needlematch $screenName, $key, 5, 60;
+
+    unless (handle_screen([keys %handlers], \%handlers, assert => 0, timeout => 60)) {
+        send_key_until_needlematch $screenName, $cmd{next}, 5, 60;
```

Comment:
I think this will be never executed as handler will exit only when $exit is true.

Reply:
`handle_screen()` exits when:
- handler function returns true (line 162 will be skipped)
- the internal loop runs out of retries, in this case after 3 or 4 iterations, depending on `$build` (line 162 will be executed)
- none of the needles match, since `assert` is 0 (line 162 will be executed)